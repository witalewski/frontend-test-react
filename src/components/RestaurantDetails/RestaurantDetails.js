import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { toJS } from 'mobx';
import { inject, observer } from 'mobx-react';
import { Status } from '../Status';
import { StarRating } from '../StarRating';
import { convertToFriendlyRoute } from '../../utils';

class RestaurantDetails extends Component {
  componentDidMount() {
    const {
      match: {
        params: { urlFriendlyName },
      },
      restaurants,
      setCurrentRestaurantId,
    } = this.props;
    const restaurant = toJS(
      restaurants.find(
        ({ name }) => convertToFriendlyRoute(name) === urlFriendlyName
      )
    );
    this.setState({
      restaurant,
    });
    setCurrentRestaurantId(restaurant.id);
  }

  getContent() {
    const { currentRestaurantDetails } = this.props;
    const {
      categories,
      name,
      image_url,
      rating,
      price,
      is_closed,
    } = this.state.restaurant;

    const category = categories.length && categories[0].title;

    return (
      <div className="restaurants-details">
        <h1>{name}</h1>
        <img alt={name} src={image_url} />
        <StarRating rating={rating} />
        <div className="restaurants-list-item__details">
          <div>
            {category} â€¢ {price}
          </div>
          <Status open={!is_closed} />
        </div>
        <div className="restaurants-list-item__learn-more-wrapper">
          <Link to={`/${convertToFriendlyRoute(name)}`}>Learn more</Link>
        </div>
      </div>
    );
  }
  render() {
    return this.state ? this.getContent() : null;
  }
}

export { RestaurantDetails };
export default inject(({ appState }) => ({
  restaurants: appState.restaurants,
  setCurrentRestaurantId: appState.setCurrentRestaurantId,
  currentRestaurantDetails: appState.currentRestaurantDetails,
}))(observer(RestaurantDetails));
