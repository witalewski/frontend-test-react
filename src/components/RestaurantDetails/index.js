export { default as RestaurantDetails } from './RestaurantDetails';
