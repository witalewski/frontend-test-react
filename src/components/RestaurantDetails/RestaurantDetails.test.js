import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import MOCK_RESTAURANT from '../../fixtures/restaurant.json';
import { RestaurantDetails } from './RestaurantDetails';

describe('RestaurantDetails', () => {
  it('matches snapshot', () => {
    const renderer = new ShallowRenderer();
    renderer.render(
      <RestaurantDetails currentRestaurantDetails={MOCK_RESTAURANT} setCurrentRestaurantId={jest.fn()} match={{params:{id:MOCK_RESTAURANT.id}}} />
    );
    const result = renderer.getRenderOutput();

    expect(result).toMatchSnapshot();
  });
});
