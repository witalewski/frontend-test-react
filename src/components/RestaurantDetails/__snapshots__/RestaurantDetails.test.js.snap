// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestaurantDetails matches snapshot 1`] = `
<div
  className="restaurants-details"
>
  <div
    className="restaurants-details__main"
  >
    <h1>
      Bagels & Friends
    </h1>
    <StarRating
      rating={4.5}
    />
    <div
      className="restaurants-details-item__details"
    >
      <div>
        Breakfast & Brunch
         • 
        $$
      </div>
      <Status
        open={true}
      />
    </div>
    <div
      className="restaurant-details__photos"
    >
      <Photos
        photos={
          Array [
            Object {
              "src": "https://maps.googleapis.com/maps/api/staticmap?center=52.399683350097,16.8932677517625&markers=52.399683350097,16.8932677517625&zoom=15&size=400x288&maptype=roadmap&key=AIzaSyCDc4LkExUhXGwButHuEENHOTE3iAJ8k2E",
              "url": "https://www.google.com/maps/search/Bagels & Friends,52.399683350097,16.8932677517625",
            },
            Object {
              "src": "https://s3-media2.fl.yelpcdn.com/bphoto/zmRMxckIj4rhutGrSRcvYA/o.jpg",
            },
            Object {
              "src": "https://s3-media1.fl.yelpcdn.com/bphoto/T5WhG8Q7cjXNeOu8OteB1Q/o.jpg",
            },
            Object {
              "src": "https://s3-media1.fl.yelpcdn.com/bphoto/vqOmlb109NiG1x54E4UAWg/o.jpg",
            },
          ]
        }
      />
    </div>
    <div
      className="restaurants-details__address"
    >
      ul. Wyspiańskiego 26, 60-751 Poznań, Poland
    </div>
  </div>
  <inject-ReviewsList />
</div>
`;
